src/senaite/core/content/sampletemplate.py

```diff
@@ -508,12 +511,29 @@ def setServices(self, value):
                 "part_id": part_id,
             })
 
+        # always keep inactive services so they come up again when reactivated
+        uids = [record.get("uid") for record in records]
+        for record in self.getRawServices():
+            uid = record.get("uid")
+            if uid in uids:
+                continue
+            obj = api.get_object_by_uid(uid)
+            if not api.is_active(obj):
+                records.append(record)
```

Comment:
I would suggest to refactor this code to allow to flush inactive services:

```python
def setServices(self, value, keep_inactive=True):
    ...
    new_uids = [record.get("uid") for record in records]
    if keep_inactive:
        to_check = filter(lambda uid: uid not in uids, self.getServiceUIDs())
        inactive = filter(lambda uid: not api.is_active, to_check)
        uids.extend(inactive)
    ...
```

Reply:
Addressed with https://github.com/senaite/senaite.core/pull/2637/commits/f12121e96ec293d7452b74d3025d601c55b0d2dd